auto_nag/bzcleaner.py

```diff
@@ -219,6 +234,70 @@ def add_auto_ni(self, bugid, data):
             }
         return True
 
+    def add_prioritized_action(
+        self, bug, quota_name, needinfo=None, autofix=None, allow_multi_ni=False
+    ):
+        assert needinfo or autofix
+
+        # Avoid having more than one ni from our bot
+        if needinfo and not allow_multi_ni and self.has_bot_set_ni(bug):
+            needinfo = autofix = None
+
+        action = {"bugid": str(bug["id"]), "needinfo": needinfo, "autofix": autofix}
+        action["key"] = self.get_sort_action_key(action, bug)
```

Comment:
Instead of adding a key here, we can call the get_sort_action_key function directly when we are calling `sort`

Reply:
Generating the key depends on the full `bug` object which we do not keep. Calling the function here will allow us to keep only the info we need for the sorting process only. The bug object can be big, including all flags, comments, history and everything else (depends on the tool).

Reply:
We are storing the bug objects anyway for generating the email in the end, so we might as well store a reference to them here too and avoid storing the key (less complexity)

Reply:
The bug objects that we store for generating the email are different:
- In the `handle_bug()` method, we usually create a new object; we do not reuse the original object:
https://github.com/mozilla/relman-auto-nag/blob/c92e238ab5e677fa807a74438a540369167f032d/auto_nag/scripts/assignee_no_login.py#L93
- This new bug object doesn't necessarily have the fields needed for the sorting process. 


Based on the simplification in c92e238ab5e677fa807a74438a540369167f032d:
- If a tool requires customizing the prioritization, it just needs to override `get_bug_sort_key()` to return the values needed to sort the bug. In this way, we keep in memory only the data needed for the sorting process instead of keeping the original bug object which could be extensive. 
- If the tool doesn't override `get_bug_sort_key()`, we will not store any key.

Reply:
Besides memory optimization, I have no problem referencing the original bug object and calling the `get_bug_sort_key()` later when we do the actual sorting process.  If you prefer dropping this optimization, I'm ok with that.

Reply:
Yeah I'd prefer dropping the optimization in favor of simplicity. It shouldn't be a big difference anyway.

Reply:
Done in 7517770b673292d046a355610b3f996b546c9ffc. Simplicity wins :)
![giphy](https://user-images.githubusercontent.com/4151357/164753064-dc2031b6-5598-49ec-9e5f-8db79af4c88c.gif)